handlers/hash_utils.go

```diff
@@ -62,26 +64,40 @@ func CalculateBasicHashes(client *AwsClient, bucket string, key string) (*HashIn
 	}
 	log.Printf("Size %d", *objectSize)
 
-	result, _ := GetS3ObjectOutput(client, bucket, key)
-	p := make([]byte, ChunkSize)
-
+	start := int64(0)
+	step := int64(ChunkSize)
 	for {
-		n, err := result.Body.Read(p)
-
-		if err != nil && err != io.EOF {
-			return nil, int64(-1), err
+		chunkRange := fmt.Sprintf("bytes: %d-%d", start, minOf(start+step, *objectSize-1))
```

Comment:
should we let end as `minOf(start+step-1, *objectSize-1))`, so that each chunk (except for the last one) will be excatly `ChunkSize` long? and we can remove `chunkLength` below

Reply:
just ignore this code. I have just realized that the code version in master branch is better